### PR TITLE
Implemented support for topology spread constraints

### PR DESCRIPTION
**Pull request by ellistarn**
This PR includes support for topologyspreadconstraints as an alternative to podantiaffinity. 

## Testing
```
cat ./testvalues.yaml && helm install vault . --values ./testvalues.yaml && kubectl get pods -oyaml | grep topology
server:
  affinity: ""
  topologySpreadConstraints:
    - maxSkew: 1
      topologyKey: "kubernetes.io/hostname"
      whenUnsatisfiable: DoNotSchedule
      labelSelector:
        matchLabels:
          app.kubernetes.io/name: vault
injector:
  affinity: ""
  topologySpreadConstraints:
    - maxSkew: 1
      topologyKey: "kubernetes.io/hostname"
      whenUnsatisfiable: DoNotSchedule
      labelSelector:
        matchLabels:
          app.kubernetes.io/name: vault-agent-injector
NAME: vault
LAST DEPLOYED: Fri Sep 17 16:52:21 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
NOTES:
Thank you for installing HashiCorp Vault!

Now that you have deployed Vault, you should look over the docs on using
Vault with Kubernetes available here:

https://www.vaultproject.io/docs/


Your release is named vault. To learn more about the release, try:

  $ helm status vault
  $ helm get manifest vault
    topologySpreadConstraints:
      topologyKey: kubernetes.io/hostname
    topologySpreadConstraints:
      topologyKey: kubernetes.io/hostname
```

## Questions

Please let me know if there are additional steps I need to take.
- Do I need to regenerate values.schema.json
- Do I need to add topology specific tests?

Make test failed for me with
```
ERROR: (gcloud.auth.activate-service-account) Unable to read file [vault-helm-test.json]: [Errno 2] No such file or directory: 'vault-helm-test.json'
make: *** [Makefile:71: acceptance] Error 1
make: *** [test-acceptance] Error 2
```

### CODE REVIEW

**Comment by ellistarn** [comment]
Hey @benashz, I'm new to this repo. Is there anything I should do to get some eyes on this PR?

**Comment by benashz** [comment]
Hey @ellistarn, it would be good if you could extend the unit tests to validate this feature. You will probably need to update `values.schema.json` as well. It's best to do this by hand, rather than rely on the make target.

File: values.yaml [hunk 1]
```diff
@@ -175,6 +175,10 @@ injector:
               component: webhook
           topologyKey: kubernetes.io/hostname
 
+  # Topology settings for injector pods
+  # ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
+  topologySpreadConstraints: []
```
Comment (benashz) [inline values.yaml:180]:
```suggestion
  # This should be either a multi-line string or YAML matching the topologySpreadConstraints array
  # in a PodSpec.
  topologySpreadConstraints: []
```

File: values.yaml [hunk 2]
```diff
@@ -430,6 +434,10 @@ server:
               component: server
           topologyKey: kubernetes.io/hostname
 
+  # Topology settings for server pods
+  # ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
+  topologySpreadConstraints: []
```
Comment (benashz) [inline values.yaml:439]:
```suggestion
  # This should be either a multi-line string or YAML matching the topologySpreadConstraints array
  # in a PodSpec.
  topologySpreadConstraints: []
```

**Comment by swenson** [comment]
This was merged as part of #652

**Pull request closed without merging**